# Convert Excel table to Markdown syntax
Because typing out the markdown code for a table is tough and tedious, therefore, I use python to help me generate the markdown syntax from excel tables. Excel tables makes it more user friendly for data entry.

In [1]:
import pandas as pd

In [2]:
df = pd.read_excel(
    "../data/excel_table_to_md.xlsx",
)
df

Person    name  height_cm  weight_kg
0       1    Jane        168         68
1       2   Peter        170         60
2       3  Edward        172         93
3       4     Tom        174         68

In [3]:
table_md = df.to_markdown()
print(table_md)


|    |   Person | name   |   height_cm |   weight_kg |
|---:|---------:|:-------|------------:|------------:|
|  0 |        1 | Jane   |         168 |          68 |
|  1 |        2 | Peter  |         170 |          60 |
|  2 |        3 | Edward |         172 |          93 |
|  3 |        4 | Tom    |         174 |          68 |


|    |   Person | name   |   height_cm |   weight_kg |
|---:|---------:|:-------|------------:|------------:|
|  0 |        1 | Jane   |         168 |          68 |
|  1 |        2 | Peter  |         170 |          60 |
|  2 |        3 | Edward |         172 |          93 |
|  3 |        4 | Tom    |         174 |          68 |

In [4]:
# This overwrite all existing content with new ones
with open("../output/output_nb_2.md", "w") as f:
    f.write(table_md)

# This appends content
with open("../output/output_nb_2.md", "a") as f:
    f.write("\n\n")  # Write two empty lines to a markdown file
    f.write(table_md)  # Continues writing from the last empty line
